dataprofiler/tests/profilers/profiler_options/test_structured_options.py

```diff
@@ -0,0 +1,173 @@
+        option.data_labeler = StructuredOptions()
+        option.multiprocess = StructuredOptions()
+
+        expected_error = set()
+        for key in self.keys:
+            ckey = key.capitalize()
+            if key == "data_labeler": ckey = "DataLabeler"
+            elif key == "category": ckey = "Categorical"
+            elif key == "datetime": ckey = "DateTime"
+            if key == "multiprocess":
+                expected_error.add('{}.{} must be a(n) BooleanOption.'.format(optpth, key, ckey))
+            else:
+                expected_error.add('{}.{} must be a(n) {}Options.'.format(optpth, key, ckey))
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, set(option._validate_helper()))
+            
+    def test_validate(self):
+        # Valid cases should return None while invalid cases should return or throw a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+    
+        # Default Configuration Is Valid
+        self.assertEqual(None, option.validate())
+        
+        # Option is_enabled is not a boolean
+        for key in self.keys:
+            option.set({'{}.is_enabled'.format(key): "Hello World"}) 
+        
+        expected_error = ["{}.{}.is_enabled must be a Boolean.".format(optpth, key) 
+            for key in self.keys]
+        expected_error = set(expected_error)
+        # Verify expected errors are a subset of all errors
+        with self.assertRaises(ValueError) as cm:
+            option.validate(raise_error=True)
+        raised_error = set(str(cm.exception).split("\n"))
+        self.assertEqual(expected_error, expected_error.intersection(raised_error))
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option.validate(raise_error=False))))
+
+        # Wrong Class Type
+        option = self.get_options()
+        option.int = StructuredOptions()
+        option.float = StructuredOptions()
+        option.datetime = StructuredOptions()
+        option.text = StructuredOptions()
+        option.order = StructuredOptions()
+        option.category = StructuredOptions()
+        option.data_labeler = StructuredOptions()
+        option.multiprocess = StructuredOptions()
+
+        expected_error = set()
+        for key in self.keys:
+            ckey = key.capitalize()
+            if key == "data_labeler": ckey = "DataLabeler"
+            elif key == "category": ckey = "Categorical"
+            elif key == "datetime": ckey = "DateTime"
+            if key == "multiprocess":
+                expected_error.add('{}.{} must be a(n) BooleanOption.'.format(optpth, key, ckey))
+            else:
+                expected_error.add('{}.{} must be a(n) {}Options.'.format(optpth, key, ckey))
+        expected_error = set(expected_error)
+        # Verify expected errors are a subset of all errors
+        self.assertSetEqual(expected_error, set(option.validate(raise_error=False)))
```

Comment:
I'm confusing why you are doing set equality here but subsets in other assertions? This seems redundant with the assertions in the following lines too

Reply:
That is a good point, let me fix it in the future assertion - I think I got used to doing the subset assertions